formlayout.py

```diff
@@ -334,15 +334,32 @@ class PushLayout(QHBoxLayout):
     """Push buttons horizontal layout"""
     def __init__(self, buttons, parent=None):
         QHBoxLayout.__init__(self)
+        self.result = parent.result
+        self.dialog = parent.get_dialog()
         for button in buttons:
             label, callback = button
             self.btn = QPushButton(label)
             if SIGNAL is None:
-                self.btn.clicked.connect(callback)
+                self.btn.clicked.connect(self.call(callback))
```

Comment:
Short version : because it doesn't works !

Because of the loop scope, you will have all the callbacks equal to the last lambda evaluation.